### PR TITLE
feat(charts): Implement discover link unfurling

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Adds discover link unfurling (into charts) behind a feature flag

### CODE REVIEW

File: src/sentry/integrations/slack/unfurl/discover.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        params = link.args["query"]
+        params.setlist("order", params.get("sort"))
+        display_mode = str(link.args["query"].get("display", "default"))
+
+        if "daily" in display_mode:
+            params.setlist("interval", ["1d"])
+        if "top5" in display_mode:
+            params.setlist("topEvents", [f"{TOP_N}"])
+
+        logging.info(params)
```
Comment (scefali) [inline src/sentry/integrations/slack/unfurl/discover.py:52]: @EvanPurkhiser why not make a local logger here so we know what file this is coming from?

Comment (EvanPurkhiser) [reply src/sentry/integrations/slack/unfurl/discover.py:52]: oh this is leftover debugging :)

File: src/sentry/integrations/slack/unfurl/discover.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    Extracts discover arguments from the discover link's query string
+    """
+    parsed_url = urlparse(url)
+    query = QueryDict(parsed_url.query).copy()
+
+    # Remove some unused query keys
+    query.pop("widths")
+    query.pop("amp")
+
+    return dict(**args, query=query)
```
Comment (scefali) [inline src/sentry/integrations/slack/unfurl/discover.py:100]: @EvanPurkhiser are you sure `query` can never be part of `args`?

Comment (EvanPurkhiser) [reply src/sentry/integrations/slack/unfurl/discover.py:100]: Yeah, args comes from the matcher's regex, which is just `org_slug` right now

File: src/sentry/integrations/slack/unfurl/discover.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        params = link.args["query"]
+        params.setlist("order", params.get("sort"))
+        display_mode = str(link.args["query"].get("display", "default"))
+
+        if "daily" in display_mode:
+            params.setlist("interval", ["1d"])
+        if "top5" in display_mode:
+            params.setlist("topEvents", [f"{TOP_N}"])
+
+        resp = client.get(
```
Comment (scefali) [inline src/sentry/integrations/slack/unfurl/discover.py:51]: @EvanPurkhiser hmmm shouldn't you catch for API errors and continue if one fails?

Comment (EvanPurkhiser) [reply src/sentry/integrations/slack/unfurl/discover.py:51]: Yeah that seems reasonable

File: src/sentry/integrations/slack/unfurl/discover.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+def map_discover_query_args(url: str, args: Mapping[str, str]) -> Mapping[str, Any]:
+    """
+    Extracts discover arguments from the discover link's query string
+    """
+    parsed_url = urlparse(url)
+    query = QueryDict(parsed_url.query).copy()
+
+    # Remove some unused query keys
+    query.pop("widths")
+    query.pop("amp")
```
Comment (markstory) [inline src/sentry/integrations/slack/unfurl/discover.py:98]: What is amp?

Comment (EvanPurkhiser) [reply src/sentry/integrations/slack/unfurl/discover.py:98]: Good question, I'm not sure what this was for.

Comment (dashed) [reply src/sentry/integrations/slack/unfurl/discover.py:106]: @EvanPurkhiser We don't use `amp` query string in the product.

Comment (EvanPurkhiser) [reply src/sentry/integrations/slack/unfurl/discover.py:106]: I'll remove. Not sure where I saw this

**Comment by EvanPurkhiser** [comment]
This now has associated tests!

Would be nice to get this in :)

File: src/sentry/integrations/slack/unfurl/discover.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            continue
+
+        params = link.args["query"]
+        params.setlist("order", params.get("sort"))
+        display_mode = str(link.args["query"].get("display", "default"))
+
+        if "daily" in display_mode:
+            params.setlist("interval", ["1d"])
+        if "top5" in display_mode:
+            params.setlist("topEvents", [f"{TOP_N}"])
```
Comment (wmak) [inline src/sentry/integrations/slack/unfurl/discover.py:50]: nit: why not just make top_n a string? 🤔

**Review by wmak — APPROVED** [review]
LGTM

**Pull request merged**